### ISSUE TITLE
port process collector 

### ISSUE THREAD

**Issue by siddontang** [pingcap/rust-prometheus#65]
So we can monitor current running process CPU, Memory, etc.

**Comment by siddontang** [comment]
We can use https://github.com/danburkert/procinfo-rs, but it only works in unix, I think it is ok.

**Comment by brunoqc** [comment]
It would be nice to have support for other platforms too, eventually.

**Comment by siddontang** [comment]
@brunoqc 

We can only focus on *nix (Linux + Mac OS X), we don't use Windows in our team.
This is optional metric, maybe you can help us support Windows or other platforms. :-)

**Comment by kamalmarhubi** [comment]
procinfo is in fact Linux only. Still a good starting point though.

**Comment by siddontang** [comment]
Thanks @kamalmarhubi 

We have not found any lib that supports all platforms, so now we only support Linux. If you find some or want to write one, please tell us :-)

**Comment by overvenus** [comment]
Unfortunately, procinfo does not compile with stable and beta rust (danburkert/procinfo-rs#10 and danburkert/procinfo-rs#12).

I would remove procinfo temporary, and will add it back once the problem is solved.

**Comment by kamalmarhubi** [comment]
I just opened https://github.com/danburkert/procinfo-rs/pull/14 which works around the overflow issue.

**Comment by overvenus** [comment]
That's cool! I will use procinfo once that is merged.

**Comment by kamalmarhubi** [comment]
https://github.com/danburkert/procinfo-rs/pull/15 just got merged, should now work on stable.
